libmscore/barline.cpp

```diff
@@ -725,6 +727,22 @@ void BarLine::read(XmlReader& e)
                   a->read(e);
                   add(a);
                   }
+            else if (tag == "Symbol") {
+                  Symbol* s = new Symbol(score());
```

Comment:
I think of a template method which might help to avoid code duplication.

Reply:
Yep. Something like
```
template <typename ElementType>
void BarLine::readClassData(XmlReader& e) {
      ElementType* elem = new ElementType(score());
      elem->setTrack(track());
      elem->read(e);
      add(elem);
      }
```
and call the method: `readClassData<Symbol>(e);` here and `readClassData<Image>(e);` below.

@dmitrio95 could verify the syntax.